opentelemetry-exporter-geneva/geneva-uploader/src/bench.rs

```diff
@@ -0,0 +1,46 @@
+#[cfg(test)]
+mod benchmarks {
+    use crate::payload_encoder::lz4_chunked_compression::lz4_chunked_compression;
+    use criterion::{BenchmarkId, Criterion, Throughput};
+    use rand::{rngs::StdRng, Rng, SeedableRng};
+    use std::hint::black_box;
+
+    fn setup_test_data(size: usize) -> Vec<u8> {
+        let mut data = vec![0u8; size];
+        let mut rng = StdRng::seed_from_u64(42);
+        rng.fill(&mut data[..]);
+        data
+    }
+
+    /*
+        - Criterion benchmarks (lz4_chunked_compression, lz4_flex backend):
+
+            | Input size  | Median time        | Throughput         |
+            |-------------|--------------------|--------------------|
+            | 1 byte      | ~711 ns            | ~1.3 MiB/s         |
+            | 1 KiB       | ~31.8 us           | ~30.6 MiB/s        |
+            | 1 MiB       | ~3.44 ms           | ~291 MiB/s         |
+        - No significant regressions or improvements detected.
+        - Machine: WSL2, Linux 6.6.36, AMD EPYC 7763 64-core (8 cores visible), 62GiB RAM, 16GiB swap.
+    */
+    #[test]
```

Comment:
This setup looks a bit unusual. Does this mean that we would be running this benchmark in the CI test? I think we only want to run this with `cargo bench`. Could we follow something similar to how we have added benchmarks in other methods?

Reply:
This is testing the crate private code, which won't be possible under `benches/*`. Let me see if this is something which can be re-exported as public specifically for benchmark, and then add it there. Else, will make it to be ignored with cargo test.

Reply:
Got it. I would suggest using `[[bench]]` target in that case and keep the benchmarks under `src/`. That way you can access private methods of the crate.

Reply:
As discussed offline, have made to be ignored by cargo test.